pkg/auth/client.go

```diff
@@ -4,35 +4,63 @@
 package auth
 
 import (
+	"context"
+
 	"github.com/absmach/magistrala"
 	authgrpc "github.com/absmach/magistrala/auth/api/grpc"
+	"github.com/absmach/magistrala/pkg/errors"
 	thingsauth "github.com/absmach/magistrala/things/api/grpc"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
 )
 
+var errHealthCheckFailed = errors.New("health check failed")
```

Comment:
You can rename this error to `errSvcNotServing = errors.New("service is not serving")` and use it directly (not as a wrapper) in cases the service is not in the right state.